Nodejs/Product/TypeScriptSourceMapReader/TypeScriptSourceMapReader.csproj

```diff
@@ -0,0 +1,106 @@
+    <DefineConstants>$(DefineConstants);TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+  <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
+  </PropertyGroup>
+  <PropertyGroup>
+    <DelaySign>true</DelaySign>
+  </PropertyGroup>
+  <Import Project="..\ProjectBefore.settings" />
+  <PropertyGroup>
+    <AssemblyAttributesFileName>..\..\..\typescriptfileversion.cspp</AssemblyAttributesFileName>
+  </PropertyGroup>
+  <PropertyGroup>
+    <AssemblyOriginatorKeyFile>FinalPublicKey.snk</AssemblyOriginatorKeyFile>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
+    <OutputPath>..\..\..\BuildOutput\Debug14.0\Binaries\TypeScriptSourceMapReader\</OutputPath>
+    <PlatformTarget>x86</PlatformTarget>
+    <DefineConstants>DEBUG;TRACE;FEATURE_AZURE_REMOTE_DEBUG;DEV14;DEV14_OR_LATER;DEV12_OR_LATER;DEV11_OR_LATER;NODEJS_TOOLS;NTVS_FEATURE_INTERACTIVEWINDOW</DefineConstants>
+    <AllowUnsafeBlocks>false</AllowUnsafeBlocks>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
+    <NoWarn>436</NoWarn>
+    <ErrorReport>prompt</ErrorReport>
+    <DebugType>full</DebugType>
+  </PropertyGroup>
+  <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
+  <ItemGroup>
+    <ManagedPreprocessFile Include="$(AssemblyAttributesFileName)">
+      <OutputFileExtension>.cs</OutputFileExtension>
+      <SuppressLineNumbers>true</SuppressLineNumbers>
+      <Defines>CSC_INVOKED=1;PPTARGET_CS=1</Defines>
+    </ManagedPreprocessFile>
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.VisualStudio.Debugger.Engine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\packages\Newtonsoft.Json.6.0.4\lib\net40\Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="System" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Base64VLQ.cs" />
+    <Compile Include="Base64VLQFormat.cs" />
+    <Compile Include="EncodedSourceMap.cs" />
+    <Compile Include="DecodedSourceMap.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="SourceMapReader.cs" />
+    <Compile Include="SourceMapReaderException.cs" />
+    <Compile Include="SourceMapSourceInfo.cs" />
+    <Compile Include="SourceMapSpan.cs" />
+    <Compile Include="SourceMapSpanMapping.cs" />
+    <Compile Include="SourceMapTextReader.cs" />
+    <Compile Include="SourceMapUrlHelper.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="FinalPublicKey.snk" />
+  </ItemGroup>
+  <Import Project="..\ProjectAfter.settings" />

```

Comment:
There are a bunch of things we can simplify here - go ahead and take a look at Npm.csproj to for an example of what's necessary and what's not: https://github.com/Microsoft/nodejstools/blob/master/Nodejs/Product/Npm/Npm.csproj